### PR TITLE
Created Linear_Search.rs

### PR DESCRIPTION
**Pull request by SubCoder1**
Added Linear Search in Rust

Fixes https://github.com/jainaman224/Algo_Ds_Notes/issues/2179

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of the file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Linear Search in Rust

#### Languages Used:
- Rust

#### Files Added:
- Linear_Search.rs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Linear_Search/Linear_Search.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    };
+    result
+} 
+fn main() {
+    let some_vec = vec![1,2,3,4,5,6,7,8,9,10]; 
+    match linear_search(&some_vec, &6) {
+        Some(val) => println!("The element is present at index: {}", val),
+        None => println!("The element is not present")
+    }; 
+}
```
Comment (jainaman224) [inline Linear_Search/Linear_Search.rs]: Add sample input and output.

File: Linear_Search/Linear_Search.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    let mut result: Option<usize> = None;
+    for (i, n) in vec.into_iter().enumerate() {
+        if n == key {
+            result = Some(i)
+        }
+    };
+    result
+} 
+fn main() {
+    let some_vec = vec![1,2,3,4,5,6,7,8,9,10];
```
Comment (jainaman224) [inline Linear_Search/Linear_Search.rs]: Take input from user.

File: Linear_Search/Linear_Search.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+fn linear_search<T: Ord>(vec: &Vec<T>, key: &T) -> Option<usize> {  
+    let mut result: Option<usize> = None;
+    for (i, n) in vec.into_iter().enumerate() {
+        if n == key {
+            result = Some(i)
+        }
+    };
+    result
+} 
+fn main() {
```
Comment (jainaman224) [inline Linear_Search/Linear_Search.rs]: Add a new line above this.

**Comment by SubCoder1** [comment]
@jainaman224 I've updated the file with the reviewed changes.

File: Linear_Search/Linear_Search.rs (hunk 1 above)
Comment (raghav-dalmia) [reply Linear_Search/Linear_Search.rs]: sample input-output should be in the last line.

**Comment by raghav-dalmia** [comment]
Also, squash your commits.

**Comment by raghav-dalmia** [comment]
Are you working on it?

**Comment by SubCoder1** [comment]
Yes, I've added input-output at the end of the code & squashed my commits as well.

**Comment by somya-kapoor** [comment]
Please squash your commits.

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by SubCoder1** [comment]
@somya-kapoor Yes, I'll fix it.

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days I am closing this issue.

**Pull request closed without merging**